src/charm.py

```diff
@@ -313,6 +333,16 @@ def internal_url(self) -> str:
         scheme = "https" if self._is_tls_ready else "http"
         return f"{scheme}://{self.hostname}:8080"
 
+    @property
+    def external_url(self) -> str:
+        """Return the external hostname to be passed to ingress via the relation."""
+        try:
+            if ingress_url := self.ingress.url:
+                return ingress_url
+        except ModelError as e:
```

Comment:
Should we add a TODO here to drop this try-except?
Iirc it's an issue only on relation removal?

Reply:
I took that from Prometheus, and there's no comment there, so I wouldn't know what issue to reference, if there's one to begin with :)